File: src/test/java/com/java/scogen/dataGenerator/EmployeeDetailsTest.java
package com.java.scogen.dataGenerator;

import com.java.scogen.SparkTestSessionGenerator;
import com.java.scogen.metaData.EmployeeNameList;
import org.apache.spark.sql.Column;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.junit.Assert;
import org.junit.Test;

public class EmployeeDetailsTest extends SparkTestSessionGenerator {

    private EmployeeNameList empNameList =  new EmployeeNameList();
    private EmployeeDetailsDataset empObject = new EmployeeDetailsDataset(getSpark());

    @Test
    public void uniqieNameTest()
    {
        Dataset<Row> empData = empObject.EmployeeTable(empNameList.setofUniqueNames()).cache();
        Assert.assertEquals(1000,empData.select("fname").distinct().count());
    }

    @Test
    public void countTest()
    {
        Dataset<Row> empData = empObject.EmployeeTable(empNameList.setofUniqueNames()).cache();
        Assert.assertEquals(1000,empData.count());
    }

    @Test
    public void ageRangeTest()
    {
        Dataset<Row> empData = empObject.EmployeeTable(empNameList.setofUniqueNames()).cache();
        Assert.assertEquals(0,empData.filter(new Column("age").$greater(60)
                .or(new Column("age").$less(18))).count());

    }
}

File: src/test/scala/com/scala/scogen/datagenerator/EmployeeDetailsTest.scala
package com.scala.scogen.datagenerator

import com.scala.scogen.BaseClassTest
import com.scala.scogen.metaData.EmployeeNameList
import org.junit.Assert

class EmployeeDetailsTest extends BaseClassTest {

  val listOfEmployees = EmployeeNameList.setofUniqueNames
  val employeeDs = new EmployeeDetailsDataset().employeeTable(spark,listOfEmployees).cache()
  import spark.implicits._

  "it" should "have 1000 unique first names" in {
    Assert.assertEquals(1000, employeeDs.select("fname").distinct.count)
  }

  "it" should "have 1000 employees in total" in {
    Assert.assertEquals(1000, employeeDs.count)
  }

  "all employees" should "have age range of 18 to 60" in {
    Assert.assertEquals(0, employeeDs.filter(($"age" > 60) || ($"age"<18)).count)
  }

}
